lib/xboxkrnl/xboxkrnl.h

```diff
@@ -208,6 +208,8 @@ typedef CONST UNICODE_STRING *PCUNICODE_STRING;
     (Entry)->Next = (ListHead)->Next; \
     (ListHead)->Next = (Entry);
 
+#define CONTAINING_RECORD(Address, Type, Field) ((Type *)((PCHAR)(Address) - (ULONG_PTR)(&((Type *)0)->Field)))
```

Comment:
I wonder if we should use https://gcc.gnu.org/onlinedocs/gcc/Offsetof.html#Offsetof

I don't remember why specifically, but the `NULL` pointer solution can fail or isn't valid C/C++ (warnings?).

Reply:
I'm not aware of problems with the `NULL` pointer solution, but I switched to `__builtin_offsetof` anyway to get rid of any issues there might be while not having to include C headers.